### DISCUSSION TITLE
ReassignProxyRequest and Path Transforms

### DISCUSSION THREAD

**Discussion by asgeirn** [dotnet/yarp#1749] [Q&A] [answered]
Hi!

I'm attempting using Unleash feature toggles to reassign proxy requests as described in https://microsoft.github.io/reverse-proxy/articles/ab-testing.html

However, the proxy route contains a PathSet tranform, and even if i reassign the proxy request to another cluster, this transform is still applied.

Is there a way to disable this transform for the request when I reassign the request to another cluster?

**Comment by Tratcher** [comment, marked as answer]
Hmm, interesting. Those transforms are associated with the route, not the cluster, so changing which cluster it's assigned to doesn't change the transforms.

You should be able to make this work by also re-assigning the route the way you reassign the cluster. If that works for you then please open an issue to add an overload of ReassignProxyRequest that accepts both route and cluster.

https://github.com/microsoft/reverse-proxy/blob/b988e7f35ffaf62c24651d7a6ed9827c71d461eb/src/ReverseProxy/Model/HttpContextFeaturesExtensions.cs#L49-L62
